### DISCUSSION TITLE
hive调度设置全局变量单引号问题

### DISCUSSION THREAD

**Discussion by CrazyBeeline** [apache/dolphinscheduler#6603] [Q&A]
我创建了一个hive工作流导入一些数据
如下图：
![image](https://user-images.githubusercontent.com/25030234/138806848-17c1702f-748f-490e-926b-da5f21a8bdac.png)

实现执行sql就会变成下面的图  多了两个单引号  
![image](https://user-images.githubusercontent.com/25030234/138806812-727f846a-dfa4-4f01-8215-41f2fd9307fd.png)

我该怎么配置参数去掉这些单引号

**Comment by UT55** [comment]
之前也遇到过sql里面传参数，自带单引号的问题，等一个结果。

**Comment by xuhuanxxx** [comment]
目前sql类型的任务好像只有局部变量值为数字且类型设置为数字时才能像这样拼接字符串。

sql里的列名表名这些可以是`abc`，也可以是abc，挤占了字符串本来啥都不需要声明的就能使用的声明方式的生存空间，为了让sql不把变量值当成列名表名，就加了个单引号，结果导致sql任务里变量的使用场景大大受限。

局部变量可以设置类型解决数字变量的问题，但是全局变量没有类型设置，把全局变量也加个类型，再根据类型判断加不加好像也不合适，

因为:
```
select 111='111' #1
select 111=111 #1

select 111+'111' #222
select 111+111 #222

定义k=qwer，select 1 from t1 where k=${k}

select  1 from t1 where k='qwer' #对
select  1 from t1 where k=qwer #错

select curdate()='20220215' #1
select curdate()=20220215 #1

select curdate()='2022-02-15' #1
select curdate()=2022-02-15 #0

```

可以看到有些场景不加引号报错（k=qwer），有些会导致结果与事实不符（2022-02-15），但有些场景不加也不影响结果。而且不加引号的话很多流程类似，仅仅因为表面列名不同的工作就可以通过设置全局变量，更方便的部署了。

建议sql类型的任务设置两种变量声明方式，比如${key}还是加引号，不影响老任务运行，$${key}/${{key}}表示不加引号。至于是不是放弃根据局部变量类型判断加不加引号的做法我就不知道了...
